### ISSUE TITLE
`manual_div_ceil` changes the evaluation count of a side-effectful divisor

### ISSUE THREAD

**Issue by cuishuang** [rust-lang/rust-clippy#17467]
### Summary

`manual_div_ceil` emits a machine-applicable suggestion that can change both the result and observable side effects when the repeated divisor expression is a function call.

### Reproducer

Code:

```rust
#![warn(clippy::manual_div_ceil)]

use std::sync::atomic::{AtomicUsize, Ordering};

static CALLS: AtomicUsize = AtomicUsize::new(0);
const VALUES: [u32; 2] = [4, 5];

fn next() -> u32 {
    VALUES[CALLS.fetch_add(1, Ordering::Relaxed)]
}

fn main() {
    let result = (100u32 + (next() - 1)) / next();
    println!("result={result}, calls={}", CALLS.load(Ordering::Relaxed));
}
```

Running `cargo clippy` suggests:

```text
help: consider using `.div_ceil()`: `100u32.div_ceil(next())`
```

The original expression prints:

```text
result=20, calls=2
```

The suggested expression prints:

```text
result=25, calls=1
```

I expected Clippy not to lint when proving that the two divisor expressions are equal would require treating a potentially side-effectful expression as  repeatable.

### Version

```text
rustc 1.99.0-nightly (6f72b5dd5 2026-07-22)
binary: rustc
commit-hash: 6f72b5dd5f82226a2773d40efea7bab941892a73
commit-date: 2026-07-22
host: aarch64-apple-darwin
release: 1.99.0-nightly
LLVM version: 22.1.8
clippy 0.1.99 (6f72b5dd5f 2026-07-22)
```

### Additional Labels

@rustbot label +I-suggestion-causes-bug

**Comment by cuishuang** [comment]
`manual_div_ceil` compares the repeated expressions with the default `SpanlessEq`, whose default is `allow_side_effects: true`. The comparisons should use `eq_expr_value`, or equivalently a `SpanlessEq` configured with `deny_side_effects()`.
